relay-server/src/processing/spans/process.rs

```diff
@@ -26,6 +31,12 @@ pub fn expand(spans: Managed<SampledSpans>) -> Managed<ExpandedSpans> {
             all_spans.extend(expanded);
         }
 
+        // Collect all the span_ids that an attachment could be linked to.
+        let span_ids: HashSet<SpanId> = all_spans
```

Comment:
nit
```suggestion
        let span_ids: BTreeSet<SpanId> = all_spans
```

>  BTreeMap may be more performant than a HashMap for small amounts of data.

https://develop.sentry.dev/ingestion/relay/relay-best-practices/#data-structures